### PR TITLE
Plugin codestyle guide

### PR DESCRIPTION
**Pull request by alexanderzobnin**
Please, merge into new branch until it finished.

### CODE REVIEW

File: codestyle-guide.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Grafana Plugins Code Style Draft
+
+## Repo name
+It's a good practice to use plugin repo name same as plugin id.  
+Plugin id has following structure:
+```
+(username|company)-(plugin name)-(datasource|app|panel)
```
Comment (DanCech) [inline codestyle-guide.md]: This needs to be `(grafana.net username or org slug)-(plugin name)-(datasource|app|panel)`

File: codestyle-guide.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+|   |-- datasource.js
+|   |-- module.js
+|   |-- plugin.json
+|   |-- query_ctrl.js
+|-- Gruntfile.js
+|-- LICENSE
+|-- package.json
+|-- README.md
+```
+
```
Comment (DanCech) [inline codestyle-guide.md]: We might want to note the option of putting a README.md for grafana/grafana.net under src/, that can be different from the github README.md in the repo root.  Also adding example screenshots might encourage people to provide them :)

**Comment by DanCech** [comment]
https://github.com/grafana/example-app

**Comment by daniellee** [comment]
This is now published to the docs site:

http://docs.grafana.org/plugins/developing/code-styleguide/

Along with some other articles on snapshots and editor mode.

I have updated the readme with links to it as well, and the plugin channel on public slack.

I forgot to add "We might want to note the option of putting a README.md for grafana/grafana.net under src/, that can be different from the github README.md in the repo root." as Dan said. Will fix that soon.

Anyway, a good first step and thinks for kicking it off Alex!

**Pull request closed without merging**